### PR TITLE
Fix #4300 - take css classes outside .opt-decoration in dark theme

### LINKED ISSUE #4300: [BUG]: `opt-decoration` css class not visible in dark/dark+ theme
### Describe the bug

In clang optimization pane, the opt-decorations on the left sidebar are visible only in light theme - see screenshots below.
I _suspect_ this came with #4203 by @RubenRBS , but might be completely off.

### Steps to reproduce

1. In any clang compiler pane add 'Optimization'
2. Under 'More'/'Settings' toggle dark/light themes.

### Expected behavior

opt-decorations should be visible, doesn't matter too much with what colors

### Reproduction link

https://godbolt.org/z/zEc54behz

### Screenshots

image

image

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by OfekShilon**

### CODE REVIEW

**Comment by OfekShilon** [comment]
I don't have permissions to ask a reviewer here: @RubenRBS can you please take a look?

**Comment by jeremy-rifkin** [comment]
Fixes  #4300

File: static/styles/themes/dark-theme.scss [hunk 1]
```diff
@@ -458,18 +458,21 @@ textarea.form-control {
 }
 
 .opt-decoration {
-    .analysis,
-    .mixed {
-        background: #fdfd96 !important;
-    }
+    width: 8px !important;
```
Comment (jeremy-rifkin) [inline static/styles/themes/dark-theme.scss:461]: I think these rules are redundant as they're in `explorer.scss` (which applies always regardless of theme)

**Review by jeremy-rifkin — APPROVED** [review]
It might be good to do the fix as
```diff
 .opt-decoration {
-    .analysis,
-    .mixed {
+    &.analysis,
+    &.mixed {
         background: #fdfd96 !important;
     }
 
-    .passed {
+    &.passed {
         background: #77dd77 !important;
     }
 
-    .missed {
+    &.missed {
         background: #ff6961 !important;
     }
 }
```
I'm of the opinion that it's good to be as specific as possible with selectors, I wouldn't be surprised if class names like analysis or whatnot got used in other contexts in the future 😄  `default-theme.scss` could be updated to look like this as well. 

PR does fix the issue so looks good to me!

File: static/styles/themes/dark-theme.scss (hunk 1 above)
Comment (RubenRBS) [reply static/styles/themes/dark-theme.scss:461]: Yup, specially layout code should be in `explorer.scss` file

**Comment by OfekShilon** [comment]
> I'm of the opinion that it's good to be as specific as possible with selectors, I wouldn't be surprised if class names like analysis or whatnot got used in other contexts in the future 😄 `default-theme.scss` could be updated to look like this as well.
> 
> PR does fix the issue so looks good to me!

Missed the comment on default-theme, applied in another commit. Sorry for the git mess - hopefully you guys squash the PR before merging.

**Comment by RubenRBS** [comment]
We do squash PRs, no worries. Thank you!

**Review by RubenRBS — APPROVED** [review]

**Comment by mattgodbolt** [comment]
This is now live

**Pull request merged**